### DISCUSSION TITLE
No overload for method 'NavigateAsync' takes 3 arguments

### DISCUSSION THREAD

**Discussion by Blarzek** [PrismLibrary/Prism#2261] [General]
Hi coders!
I have a problem after updating Prism on my project. I receive the following message:

`No overload for method 'NavigateAsync' takes 3 arguments`

The line where I receive that error is this:

`await _navigationservice.NavigateAsync("NavigationBar/MyView", _parameters, useModalNavigation: true);`

Does **INavigationService** changed? What can I do to solve this? Thank you so much for your help.

**Prism 8.0.0.1909**

**Comment by dansiegel** [comment]
> Does INavigationService changed?

As a result of our initial UWP support attempts for Prism 7 certain functionality was split from the INavigationService and surfaced through extension methods. In Prism 8 these API's were merged back into INavigationService however certain extensions were missed and not reported by the community until after Prism 8 was released.

This has already been fixed in #2235 and the changes are available in the Sponsor Connect CI Feed. 

> What can I do to solve this? 

- As a sponsor you can access the Prism CI feed via Sponsor Connect and simply use the latest greatest package
- You can copy the changes from #2235 into your project as a shim to be replaced whenever the next Prism release comes out
- You can refactor this call to use the known Uri parameter `useModalNavigation` on the individual views you would like pushed modally or just add it to the NavigationParameters to push all views modally.
